### PR TITLE
feat: add dogs status DB and populate

### PR DESCRIPTION
**Pull request by zubq**

### CODE REVIEW

File: backend/src/db/constants.ts [hunk 1]
```diff
@@ -82,6 +82,35 @@ INSERT INTO dogs(
     $12
 );
 `;
+
+export const CREATE_TABLE_DOGS_STATUS = `
+CREATE TABLE IF NOT EXISTS dogs_status (
+    id SERIAL PRIMARY KEY,
+    dog_id INT DEFAULT 0,
```
Comment (zinizhu) [inline backend/src/db/constants.ts]: make `dog_id` required by removing the default value

File: backend/src/db/constants.ts [hunk 2]
```diff
@@ -82,6 +82,35 @@ INSERT INTO dogs(
...
+    rescue_status TEXT,
```
Comment (zinizhu) [inline backend/src/db/constants.ts]: Will an enum type be better? We can define a few status options

File: backend/src/db/constants.ts [hunk 3]
```diff
@@ -82,6 +82,35 @@ INSERT INTO dogs(
...
+    adopter_foster_status TEXT,
```
Comment (zinizhu) [inline backend/src/db/constants.ts]: Similarly, an enum type seems better

File: backend/src/db/types.ts [hunk 4]
```diff
@@ -21,3 +21,15 @@ export type DogResponse = {
   dog: Dog | null;
   err: any;
 };
+
+export type DogStatus = {
+  dog_id: number;
+  rescue_status: string;
```
Comment (zinizhu) [inline backend/src/db/types.ts]: Change to enum type

File: backend/src/db/types.ts [hunk 5]
```diff
@@ -21,3 +21,15 @@ export type DogResponse = {
...
+  adopter_foster_status: string;
```
Comment (zinizhu) [inline backend/src/db/types.ts]: Change to enum type

File: backend/src/db/db.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ export const getDogById = async (
     }
   } catch (error) {
     err = error;
-    console.log("Detect error: ", error);
+    console.log("Detect error during getDogByID: ", error);
```
Comment (zinizhu) [inline backend/src/db/db.ts]:
```suggestion
    console.log("[getDogByID] error: ", error);
```

**Review by zinizhu — CHANGES_REQUESTED** [review]

File: backend/src/db/constants.ts (hunk 1 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 2 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 3 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/types.ts (hunk 4 above)
Comment (zubq) [reply backend/src/db/types.ts]: Done

File: backend/src/db/types.ts (hunk 5 above)
Comment (zubq) [reply backend/src/db/types.ts]: Done

File: backend/src/db/db.ts (hunk 6 above)
Comment (zubq) [reply backend/src/db/db.ts]: Done

File: backend/src/db/constants.ts [hunk 7]
```diff
@@ -82,6 +82,59 @@ INSERT INTO dogs(
...
+    'in contact',
+    'tagged'
+);
+`;
+
+export const CREATE_ENUM_ADOPTER_FOSTER_STATUS = `
+CREATE TYPE ADOPTER_FOSTER_STATUS AS ENUM (
+    'in need',
+    'in contact',
+    'commited'
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    'committed'
```

File: backend/src/db/constants.ts [hunk 8]
```diff
@@ -82,6 +82,59 @@ INSERT INTO dogs(
...
+CREATE TYPE RESCUE_STATUS AS ENUM (
+    'in need',
+    'in contact',
+    'tagged'
+);
+`;
+
+export const CREATE_ENUM_ADOPTER_FOSTER_STATUS = `
+CREATE TYPE ADOPTER_FOSTER_STATUS AS ENUM (
+    'in need',
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    'in_need',
```

File: backend/src/db/constants.ts [hunk 9]
```diff
@@ -82,6 +82,59 @@ INSERT INTO dogs(
...
+    'in contact',
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    'in_contact',
```

File: backend/src/db/constants.ts [hunk 10]
```diff
@@ -82,6 +82,59 @@ INSERT INTO dogs(
     $12
 );
 `;
+
+export const CREATE_ENUM_RESCUE_STATUS = `
+CREATE TYPE RESCUE_STATUS AS ENUM (
+    'in need',
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    'in_need',
```

File: backend/src/db/constants.ts [hunk 11]
```diff
@@ -82,6 +82,59 @@ INSERT INTO dogs(
...
+    'in contact',
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    'in_contact',
```

File: backend/src/db/constants.ts [hunk 12]
```diff
@@ -134,3 +195,24 @@ export const DOGS_INFO: Dog[] = [
     video_urls: ["../videos/AnimalLovers.mp4"],
   },
 ];
+
+export const DOGS_STATUS: DogStatus[] = [
+  {
+    dog_id: 1,
+    rescue_status: "in contact",
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    rescue_status: "in_contact",
```

File: backend/src/db/constants.ts [hunk 13]
```diff
@@ -134,3 +195,24 @@ export const DOGS_INFO: Dog[] = [
...
+    adopter_foster_status: "in need",
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    adopter_foster_status: "in_need",
```

File: backend/src/db/constants.ts [hunk 14]
```diff
@@ -134,3 +195,24 @@ export const DOGS_INFO: Dog[] = [
...
+    number_of_interested: 0
+  },
+  {
+    dog_id: 2,
+    rescue_status: "tagged",
+    adopter_foster_status: "in contact",
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    adopter_foster_status: "in_contact",
```

File: backend/src/db/constants.ts [hunk 15]
```diff
@@ -134,3 +195,24 @@ export const DOGS_INFO: Dog[] = [
...
+    number_of_interested: 1
+  },
+  {
+    dog_id: 3,
+    rescue_status: "tagged",
+    adopter_foster_status: "commited",
```
Comment (zinizhu) [inline backend/src/db/constants.ts]:
```suggestion
    adopter_foster_status: "committed",
```

File: backend/src/db/constants.ts (hunk 7 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 8 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 9 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 10 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 11 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 12 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 15 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 14 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

File: backend/src/db/constants.ts (hunk 13 above)
Comment (zubq) [reply backend/src/db/constants.ts]: Done

**Review by zinizhu — APPROVED** [review]
LGTM

**Pull request merged**